ros2/source/source/Dockerfile

```diff
@@ -73,26 +78,37 @@ RUN pip3 install -U \
     flake8-quotes \
     pytest \
     pytest-cov \
-    pytest-runner
+    pytest-repeat \
+    pytest-rerunfailures \
+    pytest-runner \
+    vcstool
+
+# clone and build buildtool
+ENV COLCON_WS /root/ros2/colcon
+RUN mkdir -p $COLCON_WS/src
+WORKDIR $COLCON_WS
+RUN wget https://raw.githubusercontent.com/colcon/colcon.readthedocs.org/master/colcon.repos && \
+    vcs import src < colcon.repos
+RUN ./src/colcon-core/bin/colcon build --paths src/*
+RUN . install/prefix.sh && colcon build
```

Comment:
Master is now `setup.sh` instead of `prefix.sh`. I don't think @dirk-thomas has done a release with that change yet, but when he does how about using `pip` instead of bootstraping?

Reply:
yes I have these changes locally but haven;t pushed them yet as this is not ready for review. Do you need an up-to-date working version pushed here ?

Reply:
> but when he does how about using pip instead of bootstraping?

As I use this for development, I bootstrap it for now because I almost always need features that are not yet released. We can definitely move to pip when this is more stable and thee is a one-line update path for users